### DISCUSSION TITLE
issue: Controller doesn't rerender after revalidation

### DISCUSSION THREAD

**Discussion by isRostCompany** [react-hook-form/react-hook-form#8635] [Q&A] [answered]
### Version Number

7.30.0

### Codesandbox/Expo snack

https://codesandbox.io/s/react-hook-form-validationschema-forked-qjt92f?file=/src/index.js

### Steps to reproduce

1. Fill "Value" input with value "nonNumber"
2. Click on submit button
3. Error appears to the right of "Value" input (it's correct, current value is not valid number)
4. Select "text" in "Type" select
5. Error to the right of "Value" input remains
https://user-images.githubusercontent.com/52238490/177327654-63f14220-b441-4094-8b2d-d9f991b1cc05.mp4

### Expected behaviour

Error to the right of the "Value" input should disappear as current value is valid "text"

### What browsers are you seeing the problem on?

Chrome

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by bluebill1049** [comment]
`Controller` doesn't dictate your render output, it's there just to collect, update and validate input value. In terms of the validation result, follow the debug code below

```tsx
resolver: async (data, context, options) => {
  // you can debug your validation schema here
  console.log('formData', data)
  console.log('validation result', await anyResolver(schema)(data, context, options))
  return anyResolver(schema)(data, context, options)
},
```

**Reply by isRostCompany** [reply]
okay, not sure I quite understand, but validation schema and validation result are both correct
what is not correct is that after Yup validator returned "true" (e.g. "field is valid in terms of current constraint") error didn't disappear because function, passed to Controller[ 'props' ][ 'render' ], didn't fire

**Reply by bluebill1049** [reply]
did you run the code snippet above to debug the problem?

**Reply by isRostCompany** [reply]
sorry, not quite sure where to insert that(

**Comment by Moshyfawn** [comment, marked as answer]
The re-render is isolated to the input you're changing. You want to provide the `rules={{ deps: ['opt'] }}` to your select input to trigger the re-validation of the text input when the select value changes.

See more info on `deps` prop in [the RHF docs Options section](https://react-hook-form.com/api/useform/register#options)

**Reply by isRostCompany** [reply]
yep, that fixes that!
thank you so much!

**Comment by isRostCompany** [comment]
thank you both @bluebill1049 and @Moshyfawn for your time
didn't see that part in the docs _-_

**Reply by Moshyfawn** [reply]
No worries, glad we could help!
